### PR TITLE
Feature/222 not implemented modal

### LINKED ISSUE #222: Frontend - Add Not Implemented modal
## Overview of the Feature Request
Add a 'Not Implemented' modal for features in the Dataset View Mode that are currently unavailable. For instance, the 'Publish Dataset' button is visible but not yet functional. To prevent user confusion, we will display this modal until the feature is developed in the SPA.

## What kind of user is the feature intended for?
All of them

## What inspired the request?
The new priorization of the SPA

## What existing behavior do you want changed?
None

## Any brand new behavior do you want to add to Dataverse?
None

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
We need a 'Not Implemented' modal to appear for the features of the app that haven't been implemented yet.
## Which issue(s) this PR closes:

- Closes #222 

## Suggestions on how to test this:
Install the necessary dependencies using npm install.
Build the design system with cd packages/design-system && npm run build.
Navigate back to the root directory with cd ../../.
Start Storybook with npm run storybook.
Visit [the Dataset Page in the Storybook](http://localhost:6006/?path=/story/pages-dataset--draft-with-all-dataset-permissions) to see the Not Implemented modal for actions that aren't implemented such as edit and publish.

### CODE REVIEW

File: src/sections/dataset/dataset-files/dataset-upload-files-button/DatasetUploadFilesButton.tsx [hunk 1]
```diff
@@ -4,15 +4,17 @@ import { useSession } from '../../../session/SessionContext'
 import styles from './DatasetUploadFilesButton.module.scss'
 import { useTranslation } from 'react-i18next'
 import { useDataset } from '../../DatasetContext'
+import { useNotImplementedModal } from '../../../not-implemented/NotImplementedModalContext'
 
 export function DatasetUploadFilesButton() {
```
Comment (MellyGray) [inline src/sections/dataset/dataset-files/dataset-upload-files-button/DatasetUploadFilesButton.tsx:9]: When I click the Upload Files button in the storybook I get this message. This is not happening with the rest of the buttons implementing the new modal, can you try it?

image

File: src/sections/not-implemented/NotImplementedModal.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Button, Modal } from '@iqss/dataverse-design-system'
+
+interface NotImplementedModalProps {
+  show: boolean
+  handleClose: () => void
+}
+
+export function NotImplementedModal({ show, handleClose }: NotImplementedModalProps) {
```
Comment (MellyGray) [inline src/sections/not-implemented/NotImplementedModal.tsx:8]: I think there should be a unit test to test the modal

**Review by MellyGray — CHANGES_REQUESTED** [review]
The code looks good but there are some buttons that also need the modal:

Access Dataset - I think we can leave this one without because it will be implemented soon
Publish Dataset
Link Dataset
Download files

File: src/sections/dataset/dataset-files/dataset-upload-files-button/DatasetUploadFilesButton.tsx (hunk 1 above)
Comment (ekraffmiller) [reply src/sections/dataset/dataset-files/dataset-upload-files-button/DatasetUploadFilesButton.tsx:9]: I'm not sure what happened, it worked for me - this is what I see when I click Upload Files. Can you try it again?  Maybe you are trying another Story?
Screenshot 2023-11-27 at 7 42 15 PM

**Comment by ekraffmiller** [comment]
merge conflicts resolved

**Review by MellyGray — APPROVED** [review]
Looks good!

**Pull request merged**